Caboodle/Browser/Browser.ios.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Caboodle
+{
+    public partial class Browser
+    {
+        public static async Task OpenAsync(string uri)
+        {
+            if (string.IsNullOrEmpty(uri))
+            {
+                throw new ArgumentNullException($"Uri cannot be null or empty.");
+            }
+
+            await Browser.OpenAsync(new System.Uri(uri));
+
+            return;
+        }
+
+        public static async Task OpenAsync(System.Uri uri)
+        {
+            await Task.Run(
+                () =>
+                {
+                    if (uri == null)
+                    {
+                        throw new ArgumentNullException($"Uri cannot be null.");
+                    }
+
+                    var url_native = new Foundation.NSUrl(uri.OriginalString);
+
+                    if (AlwaysUseExternal)
+                    {
+                        System.Diagnostics.Debug.WriteLine("External");
+
+                        Platform.BeginInvokeOnMainThread(
+                                () =>
+                                UIKit.UIApplication.SharedApplication.OpenUrl(url_native));
+                    }
+                    else
+                    {
+                        System.Diagnostics.Debug.WriteLine("Safari");
+                        var sfViewController = new SafariServices.SFSafariViewController(url_native, false);
+
+                        // set colors here, we can set tintcolors on iOS 10+
+                        Platform.BeginInvokeOnMainThread(
+                                () =>
+                                {
+                                    var vc = GetVisibleViewController();
+
+                                    if (sfViewController.PopoverPresentationController != null)
+                                    {
+                                        sfViewController.PopoverPresentationController.SourceView = vc.View;
+                                    }
+                                    vc.PresentViewController(sfViewController, true, null);
+                                });
+                    }
+                });
+
+            return;
+        }
+
+        private static UIKit.UIViewController GetVisibleViewController()
```

Comment:
We should move this out to Platform helpers.

I think that this is now my prefered way:

```
			UIViewController viewController = null;
            UIWindow window = UIApplication.SharedApplication.KeyWindow;
            if (window == null)
                throw new InvalidOperationException("There's no current active window");

            if(window.WindowLevel == UIWindowLevel.Normal)
                viewController = window.RootViewController;

            if (viewController == null)
            {
                window = UIApplication.SharedApplication.Windows.OrderByDescending(w => w.WindowLevel).FirstOrDefault(w => w.RootViewController != null && w.WindowLevel == UIWindowLevel.Normal);
                if (window == null)
                    throw new InvalidOperationException("Could not find current view controller");
                else
                    viewController = window.RootViewController;
            }

            while (viewController.PresentedViewController != null)
                viewController = viewController.PresentedViewController;
```